test_runner/fixtures/zenith_fixtures.py

```diff
@@ -231,10 +409,87 @@ def run(self, arguments: List[str]) -> 'subprocess.CompletedProcess[str]':
 
+    @cached_property
+    def auth_keys(self) -> AuthKeys:
+        pub = (Path(self.repo_dir) / 'auth_public_key.pem').read_bytes()
+        priv = (Path(self.repo_dir) / 'auth_private_key.pem').read_bytes()
+        return AuthKeys(pub=pub, priv=priv)
+
 
 @zenfixture
-def zenith_cli(zenith_binpath: str, repo_dir: str, pg_distrib_dir: str) -> ZenithCli:
-    return ZenithCli(zenith_binpath, repo_dir, pg_distrib_dir)
+def shared_simple_env(request: Any, port_distributor) -> Iterator[ZenithEnv]:
+    """
+    Simple Zenith environment, with no authentication and no safekeepers.
+
+    If TEST_SHARED_FIXTURES environment variable is set, we reuse the same
+    environment for all tests that use 'zenith_simple_env'. All compute nodes
+    are stopped between the tests, however.
+    """
+
+    if os.environ.get('TEST_SHARED_FIXTURES') is None:
+        # Creat the environment in the per-test output directory
+        repo_dir = os.path.join(get_test_output_dir(request), "repo")
+    else:
+        # We're running shared fixtures. Share a single directory.
+        repo_dir = os.path.join(str(top_output_dir), "shared_repo")
+        shutil.rmtree(repo_dir, ignore_errors=True)
+
+    builder = ZenithEnvBuilder(Path(repo_dir), port_distributor)
+
+    env = builder.init()
+
+    # For convenience in tests, create a branch from the freshly-initialized cluster.
+    env.zenith_cli(["branch", "empty", "main"])
+
+    yield env
+
+    # After the yield comes any cleanup code we need.
+    log.info('Starting postgres cleanup')
+    env.postgres.stop_all()
+
+    env.pageserver.stop(immediate=True)
+
+
+@pytest.fixture(scope='function')
+def zenith_simple_env(shared_simple_env: ZenithEnv) -> Iterator[ZenithEnv]:
+    """
+    A wrapper around 'shared_simple_env' that stops all the compute nodes
+    between tests
+    """
+    yield shared_simple_env
+
+    shared_simple_env.postgres.stop_all()
+
+
+@pytest.fixture(scope='function')
+def zenith_env(test_output_dir, port_distributor) -> Iterator[ZenithEnvBuilder]:
+    """
+    Fixture to create a Zenith environment for test.
+
+    To use, define 'zenith_env' fixture in your test to get access to the
+    builder object. Set properties on it to describe the environment.
+    Finally, initialize and start up the environment by calling
+    zenith_env.init().
+
+    After the initialization, you can launch compute nodes by calling
+    the functions in the 'env.postgres' factory object, stop/start the
+    nodes, etc.
+    """
+
+    # Create the environment in the test-specific output dir
+    repo_dir = os.path.join(test_output_dir, "repo")
+    builder = ZenithEnvBuilder(Path(repo_dir), port_distributor)
+
+    # Return the builder to the caller
+    yield builder
+
+    # After the yield comes any cleanup code we need. Stop all the nodes.
+    env = builder.env
+    if env:
+        log.info('Starting postgres cleanup')
+        env.postgres.stop_all()
+
+        env.pageserver.stop(immediate=True)
```

Comment:
Maybe it's a method of the builder so the contract is encapsulated in the builder?

Or even make the builder a context manager:

```
class ZenithEnvBuilder:
    def __enter__(self):
        pass
    def __exit__(self, exc_type, exc_value, traceback):
        if self.env:
            log.info('Starting postgres cleanup')
# ...
with ZenithEnvBuilder(....) as builder:
    yield builder
```